pkg/proc/native/proc_linux.go

```diff
@@ -368,16 +379,28 @@ func (dbp *Process) wait(pid, options int) (int, *sys.WaitStatus, error) {
 }
 
 func (dbp *Process) setCurrentBreakpoints(trapthread *Thread) error {
+	// wait for all threads to stop
+	for {
```

Comment:
We should try and ensure every thread is stopped before calling this to prevent it from having to do any more work. Can we move this logic elsewhere that would still be centralized?

Reply:
it's mirroring what we're doing in proc_darwin and proc_windows. So it's ugly, but at least it's consistent. That's why I was saying that IMHO we should fuse Halt and setCurrentBreakpoints for the native backend. I'd rather do it after 1.0, however.

Reply:
Ah, yeah that makes sense. There is so much room more refactoring and cleanup post-1.0 anyways, especially throughout `pkg/proc`. 

I think I would be happy to merge this, with the caveat that it will be refactored later, if we can not spread out the implementation of `Halt` while we keep it around. All the defer is doing is setting the flag on the thread struct that it is no longer running, why would we need to remove this? That should be an invariant that if we call `Halt` the thread should not be running.

Reply:
> All the defer is doing is setting the flag on the thread struct that it is no longer running, why would we need to remove this?

I'm using it to keep track of which threads haven't received sigstop yet.